opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -1863,9 +1821,71 @@ namespace Opm {
         // update wsolvent fraction for REIN wells
         this->updateWsolvent(fieldGroup, episodeIdx,  this->nupcolWellState());
 
-        return { changed_well_group, more_network_update, network_imbalance };
+        return changed_well_group;
     }
 
+
+    template<typename TypeTag>
+    std::tuple<bool, typename BlackoilWellModel<TypeTag>::Scalar>
+    BlackoilWellModel<TypeTag>::
+    updateNetworks(const bool mandatory_network_balance,
+                       DeferredLogger& deferred_logger,
+                       const bool relax_network_tolerance)
+    {
+        OPM_TIMEFUNCTION();
+        const int episodeIdx = simulator_.episodeIndex();
+        const auto& network = this->schedule()[episodeIdx].network();
+        if (!this->wellsActive() && !network.active()) {
+            return {false, 0.0};
+        }
+
+        const int iterationIdx = simulator_.model().newtonMethod().numIterations();
+        const auto& comm = simulator_.vanguard().grid().comm();
+        this->updateAndCommunicateGroupData(episodeIdx, iterationIdx, param_.nupcol_group_rate_tolerance_, deferred_logger);
```

Comment:
With this refactoring, potentially we are adding an extra `updateAndCommunicateGroupData`, is there some way to avoid doing that? Maybe calling it on a higher level function?

Reply:
Forget about the last comment. I decided to fix it in this PR anyway.